test/Microsoft.AspNetCore.SignalR.Tests/HubConnectionHandlerTestUtils/Hubs.cs

```diff
@@ -547,6 +552,51 @@ public async Task<ChannelReader<string>> LongRunningStream()
             return Channel.CreateUnbounded<string>().Reader;
         }
 
+        public ChannelReader<int> CancelableStream(CancellationToken token)
+        {
+            var channel = Channel.CreateBounded<int>(10);
+
+            Task.Run(() =>
+            {
+                _tcsService.StartedMethod.SetResult(null);
+                token.WaitHandle.WaitOne();
```

Comment:
Turn this into a TCS and await it instead. Having spent a large part of the past few weeks fighting legacy SignalR tests that use WaitHandles, I don't want to start a precedent :).

Reply:
I suggest an making an extension method that does the conversion: `WaitForCancellationAsync`